lib/spack/llnl/util/symlink.py

```diff
@@ -57,7 +57,7 @@ def _win32_junction(path, link):
         path = os.path.join(parent, path)
         path = os.path.abspath(path)
 
-    CreateHardLink(link, path)
+    CreateHardLink(link, path) # Hard Link is for files, junctions are for directories. This will only work for files
```

Comment:
The junction code path doesn't seem to use real junctions. It uses HardLinks instead which only work for files. Junctions only work for directories. This should either be renamed or reworked to use junctions for dirs and hard links for files. WDYT @johnwparent